### PR TITLE
test(TableToolbarSearch): convert tests to rtl

### LINKED ISSUE #12674: Use RTL for testing in `DataTable/__tests__/TableToolbarSearch-test.js`
Please use [the script to scaffold out new test files](https://github.com/carbon-design-system/carbon/issues/10184#issuecomment-1282894448)

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #12674 

Updates TableToolbarSearch tests to RTL

#### Changelog
Screenshot 2023-02-16 at 1 49 03 PM

**Changed**

- Updates and expands test coverage for `TableToolbarSearch` 
- Update snapshot

#### Testing / Reviewing

Ensure all tests pass and all props are covered by tests

`yarn test packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js`

### CODE REVIEW

File: packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js [hunk 1]
```diff
@@ -1,75 +1,129 @@
...
+
+      expect(container.firstChild).toHaveClass('custom-class');
+    });
+
+    it('should respect defaultExpanded prop', () => {
+      const { container } = render(<TableToolbarSearch />);
+
+      expect(container.firstChild).toHaveClass(
+        'cds--toolbar-search-container-expandable'
+      );
```
Comment (francinelucca) [inline packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js]: should this test use the `defaultExpanded` prop?

**Review by francinelucca — COMMENTED** [review]
LGTM, just one question

File: packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js (hunk 1 above)
Comment (alisonjoseph) [reply packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js]: Good catch, updated and updated to the correct class so it actually tests something. 😅

**Review by francinelucca — APPROVED** [review]
🚀

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**